src/Utilities/TaskItem.cs

```diff
@@ -75,14 +75,23 @@ public TaskItem()
         /// <summary>
         /// This constructor creates a new task item, given the item spec.
         /// </summary>
-        /// <comments>Assumes the itemspec passed in is escaped.</comments>
+        /// <comments>
+        /// Assumes the itemspec passed in is escaped.
+        /// If tratAsFilePath is set to true, it will try to fix itemSpac backslashes into slashes.
+        /// </comments>
         /// <param name="itemSpec">The item-spec string.</param>
+        /// <param name="treatAsFilePath">If item-spec string is a path or not.</param>
         public TaskItem(
-            string itemSpec)
+            string itemSpec,
+            bool treatAsFilePath = true)
         {
             ErrorUtilities.VerifyThrowArgumentNull(itemSpec);
 
-            _itemSpec = FileUtilities.FixFilePath(itemSpec);
+            if (treatAsFilePath)
+            {
+
+                _itemSpec = FileUtilities.FixFilePath(itemSpec);
+            }
```

Comment:
This isn't setting `_itemSpec` if `treatAsFilePath` is `false`.

```suggestion
            _itemSpec = treatAsFilePath ? FileUtilities.FixFilePath(itemSpec) : itemSpec;
```